File: packages/react-native-nitro-vlc/ios/HybridNitroVLCModule.swift
import Foundation

#if canImport(NitroModules) && canImport(MobileVLCKit)
import NitroModules
import MobileVLCKit

class HybridNitroVLCModule: HybridNitroVLCModuleSpec {
  func getVLCVersion() throws -> String {
    return VLCLibrary.shared().version
  }

  func getView(viewId: String) throws -> (any HybridNitroVLCViewSpec)? {
    return HybridNitroVLCView.lookup(viewId: viewId)
  }
}
#endif

File: packages/react-native-nitro-vlc/android/src/main/java/com/margelo/nitro/com/nitrovlc/HybridNitroVLCModule.kt
package com.margelo.nitro.com.nitrovlc

import org.videolan.libvlc.LibVLC

class HybridNitroVLCModule : HybridNitroVLCModuleSpec() {

  override fun getVLCVersion(): String {
    return LibVLC.version()
  }

  override fun getView(viewId: String): HybridNitroVLCViewSpec? {
    return HybridNitroVLCView.registry[viewId]?.get()
  }
}
